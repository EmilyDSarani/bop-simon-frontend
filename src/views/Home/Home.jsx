import Game from '../../components/Game/Game'
import FreePlay from '../../components/FreePlay/FreePlay'
import styles from './home.css'
import { useState } from 'react'
import { FormControlLabel, FormGroup, Switch } from '@mui/material'
import MenuBar from '../../components/MenuBar/MenuBar'

export default function Home() {
  const [gamePlay, setGamePlay] = useState(false)

  const handleChange = (event) => {
    setGamePlay(event.target.checked)
  }
  return (
    <main className={styles.home}>
      <MenuBar />
      <FormGroup sx={{ display: 'block' }}>
        <FormControlLabel
          control={
            <Switch
              checked={gamePlay}
              onChange={handleChange}
              aria-label="play mode switch"
            />
          }
          label={gamePlay ? 'Game Mode' : 'Free Play'}
        />
      </FormGroup>
      {gamePlay ? <Game /> : <FreePlay />}
    </main>
  )
}
