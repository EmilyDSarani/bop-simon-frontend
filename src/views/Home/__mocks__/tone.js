const tone = jest.createMockFromModule('tone')

export default (tone = 0) => {
  return tone(tone)
}
