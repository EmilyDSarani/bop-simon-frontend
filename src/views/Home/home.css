.home {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.controls {
  display: flex;
  flex-direction: row;
}