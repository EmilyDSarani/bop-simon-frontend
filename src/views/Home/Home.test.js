import Home from './Home'
import Game from '../../components/Game/Game'
import FreePlay from '../../components/FreePlay/FreePlay'
import React from 'react'
import * as Tone from 'tone'
import { screen, render, fireEvent } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

jest.mock('tone', () => {
  return {
    Synth: jest.fn().mockImplementation(function () {
      return {
        chain: () => ({
          toDestination: () => {},
          triggerAttackRelease: jest.fn()
        })
      }
    }),
    Limiter: function () {
      return { chain: () => {} }
    },
    AudioBuffer: jest.fn(),
    Frequency: jest.fn()
  }
})

it('properly renders the home view', () => {
  const container = render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  )
  expect(container).toMatchSnapshot()
})

it('calls onClick when c2 hex is clicked', () => {
  const handleClick = Tone.Synth.mock

  render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  )
  fireEvent.click(screen.getByLabelText(/c2/i))
  expect(handleClick.calls.length).toBe(2)
})

it('properly renders the Game component', () => {
  const container = render(
    <MemoryRouter>
      <Game />
    </MemoryRouter>
  )
  expect(container).toMatchSnapshot()
})

it('properly renders the FreePlay component', () => {
  const container = render(
    <MemoryRouter>
      <FreePlay />
    </MemoryRouter>
  )
  expect(container).toMatchSnapshot()
})
