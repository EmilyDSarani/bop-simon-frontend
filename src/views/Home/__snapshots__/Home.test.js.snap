// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly renders the FreePlay component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="gameMain"
      >
        <div
          class="App"
        >
          <div
            class="main"
          >
            <p>
              This is the Free Play Board
            </p>
            <div
              class="container"
            >
              <div
                id="c2"
              />
              <div
                id="d2"
              />
              <div
                id="e2"
              />
              <div
                id="f2"
              />
              <div
                id="g2"
              />
              <div
                id="a2"
              />
              <div
                id="b2"
              />
              <div
                aria-label="c3"
                id="c3"
              />
              <div
                id="d3"
              />
              <div
                id="e3"
              />
              <div
                id="f3"
              />
              <div
                id="g3"
              />
              <div
                id="a3"
              />
              <div
                id="b3"
              />
              <div
                id="c4"
              />
              <div
                id="d4"
              />
              <div
                id="e4"
              />
              <div
                id="f4"
              />
              <div
                id="g4"
              />
              <div
                id="a4"
              />
              <div
                id="b4"
              />
              <div
                id="c5"
              />
              <div
                id="d5"
              />
              <div
                id="e5"
              />
              <div
                id="f5"
              />
              <div
                id="g5"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="gameMain"
    >
      <div
        class="App"
      >
        <div
          class="main"
        >
          <p>
            This is the Free Play Board
          </p>
          <div
            class="container"
          >
            <div
              id="c2"
            />
            <div
              id="d2"
            />
            <div
              id="e2"
            />
            <div
              id="f2"
            />
            <div
              id="g2"
            />
            <div
              id="a2"
            />
            <div
              id="b2"
            />
            <div
              aria-label="c3"
              id="c3"
            />
            <div
              id="d3"
            />
            <div
              id="e3"
            />
            <div
              id="f3"
            />
            <div
              id="g3"
            />
            <div
              id="a3"
            />
            <div
              id="b3"
            />
            <div
              id="c4"
            />
            <div
              id="d4"
            />
            <div
              id="e4"
            />
            <div
              id="f4"
            />
            <div
              id="g4"
            />
            <div
              id="a4"
            />
            <div
              id="b4"
            />
            <div
              id="c5"
            />
            <div
              id="d5"
            />
            <div
              id="e5"
            />
            <div
              id="f5"
            />
            <div
              id="g5"
            />
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`properly renders the Game component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="gameMain"
      >
        <div
          class="App"
        >
          <div
            class="main"
          >
            <p>
              This is the Game Play Board
            </p>
            <div
              class="container"
            >
              <div
                aria-label="c2"
                id="c2"
              />
              <div
                id="d2"
              />
              <div
                id="e2"
              />
              <div
                id="f2"
              />
              <div
                id="g2"
              />
              <div
                id="a2"
              />
              <div
                id="b2"
              />
              <div
                id="c3"
              />
              <div
                id="d3"
              />
              <div
                id="e3"
              />
              <div
                id="f3"
              />
              <div
                id="g3"
              />
              <div
                id="a3"
              />
              <div
                id="b3"
              />
              <div
                id="c4"
              />
              <div
                id="d4"
              />
              <div
                id="e4"
              />
              <div
                id="f4"
              />
              <div
                id="g4"
              />
              <div
                id="a4"
              />
              <div
                id="b4"
              />
              <div
                id="c5"
              />
              <div
                id="d5"
              />
              <div
                id="e5"
              />
              <div
                id="f5"
              />
              <div
                id="g5"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="gameMain"
    >
      <div
        class="App"
      >
        <div
          class="main"
        >
          <p>
            This is the Game Play Board
          </p>
          <div
            class="container"
          >
            <div
              aria-label="c2"
              id="c2"
            />
            <div
              id="d2"
            />
            <div
              id="e2"
            />
            <div
              id="f2"
            />
            <div
              id="g2"
            />
            <div
              id="a2"
            />
            <div
              id="b2"
            />
            <div
              id="c3"
            />
            <div
              id="d3"
            />
            <div
              id="e3"
            />
            <div
              id="f3"
            />
            <div
              id="g3"
            />
            <div
              id="a3"
            />
            <div
              id="b3"
            />
            <div
              id="c4"
            />
            <div
              id="d4"
            />
            <div
              id="e4"
            />
            <div
              id="f4"
            />
            <div
              id="g4"
            />
            <div
              id="a4"
            />
            <div
              id="b4"
            />
            <div
              id="c5"
            />
            <div
              id="d5"
            />
            <div
              id="e5"
            />
            <div
              id="f5"
            />
            <div
              id="g5"
            />
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`properly renders the home view 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="home"
      >
        <div
          class="appBar MuiBox-root css-9orvkz"
        >
          <div
            class="MuiFormGroup-root css-18nkkfo-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  aria-label="login switch"
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    checked=""
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Logout
              </span>
            </label>
          </div>
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ij3ypl-MuiPaper-root-MuiAppBar-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-13vimk6-MuiToolbar-root"
            >
              <button
                aria-controls="menu-appbar"
                aria-haspopup="true"
                aria-label="account of current user"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <div
                class="MuiTypography-root MuiTypography-body1 css-1oc1ahk-MuiTypography-root"
              />
              <h3>
                sillyg00se 25pts Rank 3
              </h3>
              <button
                aria-controls="menu-appbar"
                aria-haspopup="true"
                aria-label="account of current user"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AccountCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </header>
        </div>
        <div
          class="controls"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            instructions
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiFormGroup-root css-18nkkfo-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  aria-label="play mode switch"
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    checked=""
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Game Mode
              </span>
            </label>
          </div>
        </div>
        <section
          class="gameMain"
        >
          <div
            class="App"
          >
            <div
              class="main"
            >
              <p>
                This is the Game Play Board
              </p>
              <div
                class="container"
              >
                <div
                  aria-label="c2"
                  id="c2"
                />
                <div
                  id="d2"
                />
                <div
                  id="e2"
                />
                <div
                  id="f2"
                />
                <div
                  id="g2"
                />
                <div
                  id="a2"
                />
                <div
                  id="b2"
                />
                <div
                  id="c3"
                />
                <div
                  id="d3"
                />
                <div
                  id="e3"
                />
                <div
                  id="f3"
                />
                <div
                  id="g3"
                />
                <div
                  id="a3"
                />
                <div
                  id="b3"
                />
                <div
                  id="c4"
                />
                <div
                  id="d4"
                />
                <div
                  id="e4"
                />
                <div
                  id="f4"
                />
                <div
                  id="g4"
                />
                <div
                  id="a4"
                />
                <div
                  id="b4"
                />
                <div
                  id="c5"
                />
                <div
                  id="d5"
                />
                <div
                  id="e5"
                />
                <div
                  id="f5"
                />
                <div
                  id="g5"
                />
              </div>
            </div>
          </div>
        </section>
      </main>
    </div>
    <div
      aria-hidden="true"
      class="MuiModal-root MuiModal-hidden MuiPopover-root MuiMenu-root css-6f6ff7-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiMenu-paper MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiPopover-paper css-edx3xz-MuiPaper-root-MuiMenu-paper-MuiPaper-root-MuiPopover-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-k811ry-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <a
              href="/home"
            >
              Play
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-ghqsh6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              href="/leaderboard"
            >
              Leaderboard
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-ghqsh6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              href="/about"
            >
              About Bop Simon
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
    <div
      aria-hidden="true"
      class="MuiModal-root MuiModal-hidden MuiPopover-root MuiMenu-root css-llifmg-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiMenu-paper MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiPopover-paper css-edx3xz-MuiPaper-root-MuiMenu-paper-MuiPaper-root-MuiPopover-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-k811ry-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-ghqsh6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            Log Out
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div>
    <main
      class="home"
    >
      <div
        class="appBar MuiBox-root css-9orvkz"
      >
        <div
          class="MuiFormGroup-root css-18nkkfo-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
            >
              <span
                aria-label="login switch"
                class="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  checked=""
                  class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                  type="checkbox"
                />
                <span
                  class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Logout
            </span>
          </label>
        </div>
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ij3ypl-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-13vimk6-MuiToolbar-root"
          >
            <button
              aria-controls="menu-appbar"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiTypography-root MuiTypography-body1 css-1oc1ahk-MuiTypography-root"
            />
            <h3>
              sillyg00se 25pts Rank 3
            </h3>
            <button
              aria-controls="menu-appbar"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
      </div>
      <div
        class="controls"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          instructions
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiFormGroup-root css-18nkkfo-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
            >
              <span
                aria-label="play mode switch"
                class="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  checked=""
                  class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                  type="checkbox"
                />
                <span
                  class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Game Mode
            </span>
          </label>
        </div>
      </div>
      <section
        class="gameMain"
      >
        <div
          class="App"
        >
          <div
            class="main"
          >
            <p>
              This is the Game Play Board
            </p>
            <div
              class="container"
            >
              <div
                aria-label="c2"
                id="c2"
              />
              <div
                id="d2"
              />
              <div
                id="e2"
              />
              <div
                id="f2"
              />
              <div
                id="g2"
              />
              <div
                id="a2"
              />
              <div
                id="b2"
              />
              <div
                id="c3"
              />
              <div
                id="d3"
              />
              <div
                id="e3"
              />
              <div
                id="f3"
              />
              <div
                id="g3"
              />
              <div
                id="a3"
              />
              <div
                id="b3"
              />
              <div
                id="c4"
              />
              <div
                id="d4"
              />
              <div
                id="e4"
              />
              <div
                id="f4"
              />
              <div
                id="g4"
              />
              <div
                id="a4"
              />
              <div
                id="b4"
              />
              <div
                id="c5"
              />
              <div
                id="d5"
              />
              <div
                id="e5"
              />
              <div
                id="f5"
              />
              <div
                id="g5"
              />
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
